playground/src/App.tsx

```diff
@@ -1,15 +1,51 @@
-import { useState, useEffect, useMemo } from 'react';
-import { createPortal } from 'react-dom';
+import { TraceContext, TraceView } from '@/components/playground/trace-view';
 import { Tabs, TabsList, TabsTrigger } from '@/components/ui/tabs';
 import { explorerPlugin } from '@graphiql/plugin-explorer';
 import { createGraphiQLFetcher } from '@graphiql/toolkit';
 import { GraphiQL } from 'graphiql';
+import { GraphQLSchema, buildClientSchema, getIntrospectionQuery, parse, validate } from 'graphql';
+import { useContext, useEffect, useMemo, useState } from 'react';
+import { createPortal } from 'react-dom';
 import { FaNetworkWired } from 'react-icons/fa';
 import { PiBracketsCurly } from 'react-icons/pi';
-import { TraceContext, TraceView } from '@/components/playground/trace-view';
+import { TbDevicesCheck } from 'react-icons/tb';
+import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from './components/ui/tooltip';
+import { Button } from './components/ui/button';
+import { cn } from './lib/utils';
 
-const graphiQLFetch = async (onFetch: any, ...args: any) => {
+const graphiQLFetch = async (
+  schema: GraphQLSchema | null,
+  clientValidationEnabled: boolean,
+  onFetch: any,
+  ...args: any
+) => {
   try {
+    if (schema && clientValidationEnabled) {
+      const query = JSON.parse(args[1].body as string)?.query as string;
```

Comment:
can accessing args[1] throw or is it 100% safe?

Reply:
It will exist. Refactored a bit.